spyder/api/plugins/new_api.py

```diff
@@ -781,302 +1211,444 @@ def get_command_line_options(self):
-    # This is the main widget of the dockable plugin.
-    # It needs to be a subclass of PluginMainWidget.
-    WIDGET_CLASS = None
+
+    WIDGET_CLASS: type[PluginMainWidget] = None
+    """
+    This is the main widget of the dockable plugin.
+
+    It must be a subclass of
+    :class:`spyder.api.widgets.main_widget.PluginMainWidget`.
+    """
 
     # ---- API: Optional attributes
     # -------------------------------------------------------------------------
-    # Define a list of plugins next to which we want to to tabify this plugin.
-    # Example: ['Plugins.Editor']
-    TABIFY = []
-
-    # Disable actions in Spyder main menus when the plugin is not visible
-    DISABLE_ACTIONS_WHEN_HIDDEN = True
-
-    # Raise and focus on switch to plugin calls.
-    # If False, the widget will be raised but focus will not be given until
-    # the action to switch is called a second time.
-    RAISE_AND_FOCUS = False
-
-    # Whether the plugin can handle file actions.
-    # If set to true, then the `create_new_file`, `open_last_closed_file`,
-    # `save_file`, `save_file_as`, `save_copy_as`, `save_all`, `revert_file`,
-    # `close_file` and `close_all` functions will be called to handle the
-    # corresponding actions. Individual actions can be disabled with
-    # `enable_file_action` in the Applications plugin.
-    CAN_HANDLE_FILE_ACTIONS = False
-
-    # List of file extensions which the plugin can open.
-    # If the user opens a file with one of these extensions, then the file
-    # will open in this plugin using the `open_file` function.
-    # Example: ['.ipynb'] for spyder-notebook
-    FILE_EXTENSIONS = []
-
-    # Whether the plugin can handle edit actions.
-    # If set to True, then the `undo`, `redo`, `cut`, `copy`, `paste` and
-    # `select_all` functions will be called to handle the corresponding
-    # actions. Individual actions can be disabled with `enable_edit_action` in
-    # the Applications plugin.
-    CAN_HANDLE_EDIT_ACTIONS = False
-
-    # Whether the plugin can handle search actions.
-    # If set to True, then the `find`, `find_next`, `find_previous` and
-    # `replace` functions will be called to handle the corresponding
-    # actions. Individual actions can be disabled with `enable_search_action`
-    # in the Applications plugin.
+
+    TABIFY: list[str] = []
+    """
+    Define a list of plugins next to which we want to to tabify this plugin.
+
+    Examples
+    --------
+
+    .. code-block:: python
+
+        TABIFY = ["Plugins.Editor"]
+    """
+
+    DISABLE_ACTIONS_WHEN_HIDDEN: bool = True
+    """
+    Disable the plugin's actions in the main menu when the plugin is hidden.
+
+    If ``True``, disable the plugin's actions when it is not shown (default);
+    if ``False``, keep them enabled.
+    """
+
+    RAISE_AND_FOCUS: bool = False
+    """
+    Give the plugin focus on switch to plugin calls.
+
+    If ``True``, the plugin will be given focus when switched to.
```

Comment:
```suggestion
    If ``True``, the plugin will be raised and given focus when switched to.
```

Reply:
Same as above, this does not appear to be correct according to either the code nor testing.